### PR TITLE
Adding ChatSession.onConnectionLost event subscription

### PR DESCRIPTION
**Pull request by mliao95**
*Issue #, if available:*
#116 

*Description of changes:*
Adding `chatSession.onConnectionLost` method which subscribes to the `CHAT_EVENTS.CONNECTION_LOST` event.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by jiahaoyu6666 — APPROVED** [review]

**Comment by spencerlepine** [comment]
Does this need new `dist/` bundles generated from npm run release, or version+CHANGELOG? Or just include in the release from the upcoming PR?: https://github.com/amazon-connect/amazon-connect-chatjs/pull/139

**Comment by mliao95** [comment]
> Does this need new `dist/` bundles generated from npm run release, or version+CHANGELOG? Or just include in the release from the upcoming PR?: #139

Just updated PR to includes `dist/` generated files.  Let's add this change to your version/CHANGELOG since it seems like a pretty minor change by itself.

**Review by spencerlepine — APPROVED** [review]

File: src/index.d.ts [hunk 1]
```diff
@@ -240,6 +240,14 @@ declare namespace connect {
      */
     onEnded(handler: (event: ChatEndedEvent) => void): void;
 
+    /**
+     * Subscribes an event handler that triggers when the session connection is lost.
+     * @param handler The event handler.
+     */
+    onConnectionLost(
+      handler: (event: ChatConnectionEstablishedEvent) => void
```
Comment (marcogrcr) [inline src/index.d.ts:248]: Should this be a `ChatConnectionLostEvent`?

**Review by marcogrcr — COMMENTED** [review]
Thanks for making this change!

File: README.md [hunk 2]
```diff
@@ -569,6 +569,18 @@ chatSession.onAutoDisconnection(event => {
 ```
 `onParticipantIdle`, `onParticipantReturned`, and `onAutoDisconnection` are related to [set up chat timeouts for chat participants](https://docs.aws.amazon.com/connect/latest/adminguide/setup-chat-timeouts.html).
 
+#### `chatSession.onConnectionLost()`
+```js
+chatSession.onConnectionLost(event => {
+  const { chatDetails, data } = event;
+  switch (data.ContentType) {
```
Comment (marcogrcr) [inline README.md:576]: I believe `data` here will always have the value `{}`. So `data.ContentType` will be `undefined`.

**Pull request merged**